hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/workflows/Context.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.spi.workflows;
+
+import com.hedera.hapi.node.base.HederaFunctionality;
+
+public interface Context {
```

Comment:
Nice idea! 🥂 

What about naming `OperationContext` to add more domain meaning? (I generally use HAPI "operations" as am umbrella term to include both transactions and queries.)